src/main/java/com/google/cloud/tools/managedcloudsdk/install/Extractor.java

```diff
@@ -42,24 +44,60 @@
     this.messageListener = messageListener;
   }
 
-  /** Extract and return a {@link Path} to the Cloud SDK home directory. */
-  @Override
-  public Path call() throws IOException {
+  /** Extract and return a {@link Path} to the extraction root. */
+  public Path extract() throws IOException, InterruptedException {
     messageListener.message("Extracting archive: " + archive);
 
-    extractorProvider.extract(archive, destination, messageListener);
+    try {
+      extractorProvider.extract(archive, destination, messageListener);
+    } catch (IOException ex) {
+      try {
+        messageListener.message("Extraction failed, cleaning up " + destination);
+        cleanUp(destination);
+      } catch (Exception exx) {
+        messageListener.message("Failed to cleanup directory");
+      }
+      // intentional rethrow after cleanup
+      throw ex;
+    }
 
-    // this is convention
-    Path cloudSdkHome = destination.resolve("google-cloud-sdk");
-    if (!Files.isDirectory(cloudSdkHome)) {
-      throw new FileNotFoundException(
-          "After extraction, Cloud SDK home not found at " + cloudSdkHome);
+    // we do not allow interrupt mid extraction, so catch it here, we still end up
+    // with a valid directory though, so don't clean it up.
+    if (Thread.currentThread().isInterrupted()) {
+      messageListener.message("Process was interrupted");
+      throw new InterruptedException("Process was interrupted");
     }
-    return cloudSdkHome;
+
+    return destination;
   }
 
   @VisibleForTesting
   ExtractorProvider getExtractorProvider() {
     return extractorProvider;
   }
+
+  private void cleanUp(final Path target) throws IOException {
+    Preconditions.checkArgument(Files.isDirectory(target));
+    Files.walkFileTree(
```

Comment:
What about using the google commons function? https://google.github.io/guava/releases/21.0/api/docs/com/google/common/io/MoreFiles.html#deleteDirectoryContents-java.nio.file.Path-com.google.common.io.RecursiveDeleteOption...-

Reply:
Needs java8. Will add note though.